ta4j/src/main/java/org/ta4j/core/indicators/pivotpoints/FibonacciReversalIndicator.java

```diff
@@ -35,62 +36,46 @@
  */
 public class FibonacciReversalIndicator extends RecursiveCachedIndicator<Decimal> {
 
-    private PivotPointIndicator pivotPointIndicator;
-    private PivotLevel level;
-
-    public enum PivotLevel{
-        RESISTANCE_3(Decimal.ONE, true),
-        RESISTANCE_2(Decimal.valueOf(0.618), true),
-        RESISTANCE_1(Decimal.valueOf(0.382), true),
-        SUPPORT_1(Decimal.valueOf(0.382), false),
-        SUPPORT_2(Decimal.valueOf(0.618), false),
-        SUPPORT_3(Decimal.ONE, false);
-
-        private Decimal fibFactor;
-        private boolean isResistance;
-
-        PivotLevel(Decimal fibFactor, boolean isResistance){
-            this.fibFactor = fibFactor;
-            this.isResistance = isResistance;
-        }
-
-        public Decimal getFactor(){
-            return this.fibFactor;
-        }
-
-        public boolean isResistance(){
-            return isResistance;
-        }
+    private final PivotPointIndicator pivotPointIndicator;
+    private final FibReversalTyp fibReversalTyp;
+    private final Decimal fibonacciFactor;
 
+    public enum FibReversalTyp{
```

Comment:
Rename this to "PivotType" or "PivotReversalType" and put this enum-declaration within "PivotPointIndicator". With this you can reuse "PivotType" also for other "XPivotPointIndicator" (for example, DeMarkPivotIndicator). No need to duplicate code.